rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -188,6 +189,17 @@ rmw_create_publisher(
     return nullptr;
   }
 
+  switch(qos_policy) {
+    case BEST_EFFORT:
+      datawriter_qos.reliability.kind = DDS::BEST_EFFORT_RELIABILITY_QOS;
+      break;
+    case RELIABLE:
+      datawriter_qos.reliability.kind = DDS::RELIABLE_RELIABILITY_QOS;
+      break;
+    default:
+      break;
```

Comment:
This case should throw a `runtime_error` so that it is easy to catch if more qos policies are added but not everywhere handled.

Same below.